### PR TITLE
Add a global site notice

### PR DESCRIPTION
**Pull request by Badlapje**
Prior to this change, there was no way to alert the user with temporary content.  For example: what changed in the last release.

This change adds a global site notice.  It uses two parameters to set the allowed html & whether or not to display the notice.  The content of the notice itself can be set in the translation.

Pivotal ticket at: https://www.pivotaltracker.com/story/show/176902060

### CODE REVIEW

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/SiteNoticeController.php [hunk 1]
```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * Copyright 2017 SURFnet B.V.
```
Comment (MKodde) [inline src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/SiteNoticeController.php:4]: Nit: ~~2017~~ 2021

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/SiteNoticeController.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+     * This desire to not see the notification is stored both in the
+     * session and in cookies.
+     *
+     * @Method("GET")
+     * @Security("has_role('ROLE_USER')")
+     * @Template()
+     */
+    public function showGlobalSiteNoticeAction()
+    {
+        $date = $this->getParameter('global.site_notice.date');
```
Comment (MKodde) [inline src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/SiteNoticeController.php:44]: I would like to see the parameters DI'ed into the controller. See the EntityDetailController for an example if you need one.

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/SiteNoticeController.php [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+        $hasBeenClosed = (bool)$this->get('session')->get('site_notice.closed.' . $date);
```
Comment (MKodde) [inline src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/SiteNoticeController.php:45]: Consider injecting the session into the Controller. I see we used `this->get` on other controllers, but this is discouraged as we prefer dependency injection over grabbing stuff from the container.

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Twig/GlobalSiteNotice.php [hunk 4]
```diff
@@ -0,0 +1,75 @@
+<?php
+
+/**
+ * Copyright 2017 SURFnet B.V.
```
Comment (MKodde) [inline src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Twig/GlobalSiteNotice.php:4]: Same nit as before :beetle:

**Review by MKodde — CHANGES_REQUESTED** [review]
Nice work again. Some things might need your attention.In addition to the code review actions you might want to update the story according to the new insights, requirements and improvements that have been implemented. 

This request was triggered by my limited understanding of the notice.date component that was only hinted on in the Pivotal story. Is this date added a 'show from this date and forward' date? Or show until, or ... In code I see the date added to the session.

And finally, you mentioned the closing trigger will also be stored in cookie. But I do not see the creation of an explicit cookie other than the session cookie. Is that also intentional?

This might need some additional documentation.

Disclaimer: I have not tested it in my browser yet.

File: src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/SiteNoticeController.php (hunk 3 above)
Comment (Badlapje) [reply src/Surfnet/ServiceProviderDashboard/Infrastructure/DashboardBundle/Controller/SiteNoticeController.php]: quite a few lines of code removed thanks to this suggestion: nice one!

**Comment by Badlapje** [comment]
> Nice work again. Some things might need your attention.In addition to the code review actions you might want to update the story according to the new insights, requirements and improvements that have been implemented.

I agree there, but i'm awaiting feedback from @teunfransen.  Saves me having to edit it twice ;-)

**Review by MKodde — APPROVED** [review]

**Pull request merged**